Source/WebCore/editing/MarkupAccumulator.cpp

```diff
@@ -741,4 +738,37 @@ void MarkupAccumulator::appendNonElementNode(StringBuilder& result, const Node&
     }
 }
 
+static bool isElementExcludedByRule(const MarkupExclusionRule& rule, const Element& element)
+{
+    if (!rule.elementLocalName.isNull() && !equalIgnoringASCIICase(rule.elementLocalName, element.localName()))
+        return false;
+
+    unsigned matchedAttributes = 0;
+    if (element.hasAttributes()) {
+        for (auto& [attributeLocalName, attributeValue] : rule.attributes) {
+            if (attributeLocalName.isNull()) {
+                ++matchedAttributes;
+                continue;
+            }
+
+            for (const Attribute& attribute : element.attributesIterator()) {
```

Comment:
This is O(n^2). We should consider using a HashMap when the number of attributes is above certain threshold.
We can probably just add a FIXME for now.